apps/qubit/modules/menu/actions/editAction.class.php

```diff
@@ -40,9 +40,13 @@ protected function addField($name)
     {
       case 'name':
       case 'path':
-        $this->form->setDefault($name, $this->menu[$name]);
-        $this->form->setValidator($name, new sfValidatorString(array('required' => true)));
-        $this->form->setWidget($name, new sfWidgetFormInput);
+        // Don't allow locked menus to be renamed
+        if ($name != 'name' || !$this->menu->isProtected())
```

Comment:
This condition is a little hard to understand at first glance. Not a big deal but I think it's clearer with:

```
if ($name = 'name' && $this->menu->isProtected())
{
  break;
}
...
```